icu4j/main/core/src/main/java/com/ibm/icu/util/IslamicCalendar.java

```diff
@@ -582,28 +579,88 @@ private final static boolean civilLeapYear(int year)
         return (14 + 11 * year) % 30 < 11;
     }
 
+    /**
+     * The yearStart function for Islamic Civil calendar.
+     */
+    private long yearStartIslamicCivil(int year) {
+        return (year-1)*354 + (long)Math.floor((3+11*year)/30.0);
+    }
+
+    /**
+     * The yearStart function for Islamic calendar.
+     */
+    private long yearStartIslamic(int year) {
+        return trueMonthStart(12*(year-1));
+    }
+
+    /**
+     * The yearStart function for Islamic Umalqura calendar.
+     */
+    private long yearStartIslamicUmalqura(int year) {
+        if (year < UMALQURA_YEAR_START || year > UMALQURA_YEAR_END) {
+            return yearStartIslamicCivil(year);
+        }
+        year -= UMALQURA_YEAR_START;
+        // rounded least-squares fit of the dates previously calculated from UMALQURA_MONTHLENGTH iteration
+        int yrStartLinearEstimate = (int)((354.36720 * year) + 460322.05 + 0.5);
+        // need a slight correction to some
+        return yrStartLinearEstimate + UMALQURA_YEAR_START_ESTIMATE_FIX[year];
+    }
+
     /**
      * Return the day # on which the given year starts.  Days are counted
      * from the Hijri epoch, origin 0.
      */
     private long yearStart(int year) {
-        long ys = 0;
-        if (cType == CalculationType.ISLAMIC_CIVIL
-                || cType == CalculationType.ISLAMIC_TBLA
-                || (cType == CalculationType.ISLAMIC_UMALQURA && (year < UMALQURA_YEAR_START || year > UMALQURA_YEAR_END))) {
-            ys = (year-1)*354 + (long)Math.floor((3+11*year)/30.0);
-        } else if(cType == CalculationType.ISLAMIC) {
-            ys = trueMonthStart(12*(year-1));
-        } else if(cType == CalculationType.ISLAMIC_UMALQURA){
-            year -= UMALQURA_YEAR_START;
-            // rounded least-squares fit of the dates previously calculated from UMALQURA_MONTHLENGTH iteration
-            int yrStartLinearEstimate = (int)((354.36720 * year) + 460322.05 + 0.5);
-            // need a slight correction to some
-            ys = yrStartLinearEstimate + UMALQURA_YEAR_START_ESTIMATE_FIX[year];
+        switch (cType) {
+            case ISLAMIC_UMALQURA:
+                return yearStartIslamicUmalqura(year);
+            case ISLAMIC:
+                return yearStartIslamic(year);
+            case ISLAMIC_CIVIL:
+            case ISLAMIC_TBLA:
+                return yearStartIslamicCivil(year);
+            default:
+                return 0;
         }
-        return ys;
     }
 
+    /**
+     * The monthStart function for Islamic Civil calendar.
+     */
+    private long monthStartIslamicCivil(int year, int month) {
```

Comment:
Same naming issue as with the year functions.  How about `monthStartForIslamicCivil()`, etc.?

Reply:
totally changed now